File: call-management/call-routing/create-ivr-menu/code-samples/createIVRMenu.js
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// PATH PARAMETERS
const accountId = '<ENTER VALUE>';

// POST BODY
const body = {
    id: '<ENTER VALUE>',
    uri: '<ENTER VALUE>',
    name: '<ENTER VALUE>',
    extensionNumber: '<ENTER VALUE>',
    prompt: {
        mode: 'Audio',
        audio: {
            uri: '<ENTER VALUE>',
            id: '<ENTER VALUE>',
            name: '<ENTER VALUE>',
            localeCode: '<ENTER VALUE>'
        },
        text: '<ENTER VALUE>',
        language: {
            uri: '<ENTER VALUE>',
            id: '<ENTER VALUE>'
        }
    },
    actions: [
        {
            input: '<ENTER VALUE>',
            action: 'Connect',
            extension: {
                uri: '<ENTER VALUE>',
                id: '<ENTER VALUE>'
            },
            phoneNumber: '<ENTER VALUE>'
        },
    ]
};

const SDK = require('ringcentral');
const rcsdk = new SDK({server: process.env.serverURL, appKey: process.env.clientId, appSecret: process.env.clientSecret});
const platform = rcsdk.platform();
await platform.login({ username: process.env.username, extension: process.env.extension, password: process.env.password });
const r = await platform.post(`/restapi/v1.0/account/${accountId}/ivr-menus`, body);

File: call-management/call-routing/create-ivr-menu/code-samples/createIVRMenu.cs
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// PATH PARAMETERS
string accountId = "<ENTER VALUE>";

// POST BODY
IVRMenuInfo iVRMenuInfo = new IVRMenuInfo {
    id = "<ENTER VALUE>",
    uri = "<ENTER VALUE>",
    name = "<ENTER VALUE>",
    extensionNumber = "<ENTER VALUE>",
    prompt = new IVRMenuPromptInfo {
        mode = 'Audio',
        audio = new PromptLanguageInfo {
            uri = "<ENTER VALUE>",
            id = "<ENTER VALUE>",
            name = "<ENTER VALUE>",
            localeCode = "<ENTER VALUE>"
        },
        text = "<ENTER VALUE>",
        language = new AudioPromptInfo {
            uri = "<ENTER VALUE>",
            id = "<ENTER VALUE>"
        }
    },
    actions = new[] {
        new IVRMenuActionsInfo {
            input = "<ENTER VALUE>",
            action = 'Connect',
            extension = new IVRMenuExtensionInfo {
                uri = "<ENTER VALUE>",
                id = "<ENTER VALUE>"
            },
            phoneNumber = "<ENTER VALUE>"
        },
    }
};

RestClient rc = new RestClient(
    Environment.GetEnvironmentVariable("clientId"),
    Environment.GetEnvironmentVariable("clientSecret"),
    false
);
await rc.Authorize(
    Environment.GetEnvironmentVariable("username"),
    Environment.GetEnvironmentVariable("extension"),
    Environment.GetEnvironmentVariable("password")
);
var r = await rc.Restapi().Account(accountId).IvrMenus().Post(iVRMenuInfo);
// PROCESS RESPONSE
